client/lib/plans/index.js

```diff
@@ -67,7 +76,27 @@ export function addCurrentPlanToCartAndRedirect( sitePlans, selectedSite ) {
 }
 
 export function getCurrentPlan( plans ) {
-	return find( plans, { currentPlan: true } );
+	const currentPlan = find( plans, { currentPlan: true } );
+
+	if ( currentPlan ) {
+		return currentPlan;
+	}
+
+	// the site doesn't have defined a current plan.
+	// Let's dive in its `plan` property
+	const site = sitesList.getSelectedSite();
+
+	if ( ! site || ! site.plan ) {
+		return null;
+	}
+
+	// let's check if the current plan is the old `Enterprise` plan.
+	// if it is then let's assign `business` plan.
```

Comment:
Yes, I think it can work and its simplest. When the site does't have defied its current plant in its plans list then let's return the `plan` object gotten from `site` object.